src/vs/workbench/contrib/notebook/common/notebookExecutionStateService.ts

```diff
@@ -40,10 +41,16 @@ export interface ICellExecutionStateChangedEvent {
 	affectsNotebook(notebook: URI): boolean;
 }
 export interface INotebookFailStateChangedEvent {
-	failed: boolean;
+	visible: boolean;
 	notebook: URI;
 }
 
+export interface IFailedCellInfo {
+	cellHandle: number;
+	disposable?: IDisposable;
```

Comment:
I like this pattern, but `disposable` should not be optional because it always exists on a real info object. There is just a line while you are setting up the object where it doesn't exist, but you should be able to get rid of that